ENIGMAsystem/SHELL/Graphics_Systems/OpenGL3/GL3vertex.cpp

```diff
@@ -15,39 +17,191 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
-#include "../General/OpenGLHeaders.h"
-#include <vector>
-#include <map>
-#include <utility>
-using std::vector;
-using std::pair;
+#include "GL3profiler.h"
+#include "GL3shader.h"
+#include "GLSLshader.h"
 
-#include "../General/GSvertex.h"
-#include "../General/GStextures.h"
+#include "Graphics_Systems/General/GSvertex_impl.h"
+#include "Graphics_Systems/General/GScolor_macros.h"
+#include "Graphics_Systems/General/GSmatrix.h"
+#include "Graphics_Systems/General/GSmath.h"
+
+#include "Graphics_Systems/General/OpenGLHeaders.h"
+
+#include "Bridges/General/GL3Context.h" //Needed to get if bound texture == -1
+
+#define bind_array_buffer(vbo) if (enigma::bound_vbo != vbo) glBindBuffer( GL_ARRAY_BUFFER, enigma::bound_vbo = vbo );
```

Comment:
can we change this to a function?

Reply:
This was another copy and paste of Harri's code, iirc it was a function in the OGL Manager in `Bridges/General/` but I don't want to mess with this right now. I will take care of it probably when I clean up the bridges.